Web view/Web app/src/Institution VIew/AddLecturer/addLecturer.js

```diff
@@ -1,7 +1,42 @@
 import React from "react";
 import InstitutionNavigation from "../Navigation/InstitutionNavigation";
+import { createLecturer,deleteLecturer} from "../../graphql/mutations";
+import { useState } from "react";
+import  {API} from 'aws-amplify';
 
 const AddLecturer = () => {
+
+    const [firstName,setFirstName]=useState("")
+    const [lastName,setLastName]=useState("")
+    const [email,setEmail]= useState("")
+    const [moduleCode,setModuleCode]=useState("")   
+    const add=  async(event) => { 
+        event.preventDefault()
+        console.log("Add lecturer mutation")
+        let lecturer={ 
+            institutionId:"UP123456789",
+            firstName:firstName,
+            lastName:lastName,
+            userRole:"Lecturer",
+            email:email,
+            institution:{
+                id:"UP123456789",
+            }
+        }
+        try{
+           let mut=await API.graphql({
+              query: createLecturer,
+              variables:{input : lecturer},
+              authMode:'AMAZON_COGNITO_USER_POOLS',
+            }
+        )
+        
+            console.log("Successfully added "+lecturer.firstName)
+        }catch(e){
+            console.log(e)
```

Comment:
I do not think just logging the error to the console client side like this is safe.

The error is already being logged to the cloud via Amazon CloudWatch Logs, but I think here, the error needs to be a bit more detailed so that the frontend can display it accordingly and give the user actions to take accordingly.

Reply:
Noted. The logs  statements will be removed shortly